src/formats/MMTF.cpp

```diff
@@ -238,172 +248,259 @@ void MMTFFormat::read(Frame& frame) {
+                atom.name()
+            );
+        }
+        group.atomNameList.emplace_back(atom.name().substr(0, 5));
 
-        auto groupId = res_opt->id();
-        structure_.groupIdList.emplace_back(groupId ? *groupId : 0);
-        structure_.groupList.emplace_back();
-        structure_.groupList.back().groupName = prev_residue->name();
-        structure_.groupList.back().chemCompType =
-            prev_residue->get("composition_type") &&
-            prev_residue->get("composition_type")->kind() == Property::STRING ?
-            prev_residue->get("composition_type")->as_string() :
-            "other";
+        if (atom.type().size() > 3) {
+            warning("MMTF Writer",
+                "atom type '{}' is too long for MMTF format, it will be truncated",
+                atom.type()
+            );
+        }
+        group.elementList.emplace_back(atom.type().substr(0, 3));
 
-        auto& group = structure_.groupList.back();
-        group.formalChargeList.emplace_back(frame[i].charge());
-        group.atomNameList.emplace_back(frame[i].name());
-        group.elementList.emplace_back(frame[i].type());
+        new_atom_indexes_[i] = static_cast<int32_t>(structure_.xCoordList.size());
+        structure_.atomIdList.emplace_back(atomSkip_ + i + 1);
+        structure_.xCoordList.emplace_back(positions[i][0]);
+        structure_.yCoordList.emplace_back(positions[i][1]);
+        structure_.zCoordList.emplace_back(positions[i][2]);
     }
+    structure_.groupList.emplace_back(std::move(group));
+}
 
-    mmtf::BondAdder bondadd(structure_);
-    auto& bonds = topology.bonds();
-    auto& bond_orders = topology.bond_orders();
-    for (size_t i = 0; i < bonds.size(); ++i) {
-
-        int8_t bo;
-        switch(bond_orders[i]) {
-            case Bond::BondOrder::SINGLE:
-                bo = 1;
-                break;
-            case Bond::BondOrder::DOUBLE:
-                bo = 2;
-                break;
-            case Bond::BondOrder::TRIPLE:
-                bo = 3;
-                break;
-            case Bond::BondOrder::QUADRUPLE:
-                bo = 4;
-                break;
-            default:
-                bo = 1;
-                break;
-        }
+int8_t bond_order_to_mmtf(Bond::BondOrder order) {
+    switch(order) {
+    case Bond::BondOrder::SINGLE:
+        return 1;
+    case Bond::BondOrder::DOUBLE:
+        return 2;
+    case Bond::BondOrder::TRIPLE:
+        return 3;
+    case Bond::BondOrder::QUADRUPLE:
+        return 4;
+    case Bond::BondOrder::UNKNOWN:
+        return -1;
+    case Bond::BondOrder::QINTUPLET:
+    case Bond::BondOrder::AMIDE:
+    case Bond::BondOrder::AROMATIC:
+    case Bond::BondOrder::UP:
+    case Bond::BondOrder::DOWN:
+    case Bond::BondOrder::DATIVE_L:
+    case Bond::BondOrder::DATIVE_R:
+        warning("MMTF Writer", "bond order '{}' can not be represented in MMTF, defaulting to single bond");
```

Comment:
Is this missing an argument? Possibly `order`?

Also, I see that `default` isn't included here, but maybe it shouldn't be as it is better to explicitly list the other cases.

Reply:
Whoop, you are right.